### DISCUSSION TITLE
Is there any list for pseudo-classes

### DISCUSSION THREAD

**Discussion by Inerska** [AvaloniaUI/Avalonia#6178] [Q&A] [answered]
Hello, I am looking for a list of available TextBox pseudoclasses, I've been looking through the source code, but I cannot find it, I just want to detect when the user clicks on a text box.

Thanks in advance.

**Comment by maxkatz6** [comment, marked as answer]
It defined with an attribute https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls/TextBox.cs#L23
Also this attribute is part of API docs http://reference.avaloniaui.net/api/Avalonia.Controls/TextBox/

Same about other controls

**Reply by maxkatz6** [reply]
Although need to mention it does not includes inherited pseudoclasses.
I.e. there are some pseudoclasses in base type InputElement https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Input/InputElement.cs#L19

**Comment by FoggyFinder** [comment]
> Hello, I am looking for a list of available TextBox pseudoclasses, I've been looking through the source code, but I cannot find it, I just want to detect when the user clicks on a text box.

I doubt that such pseudo-class exists. From my understanding this attributes are about "_state_" and not about "_action_". I.e `focused`, `pointer-over`,...

"**Click**" is an event not state. So for this task using triggers seems like the right way to go. Take a look at [AvaloniaBehaviors](https://github.com/wieslawsoltes/AvaloniaBehaviors) project. There is sample for [EventTriggerBehavior](https://github.com/wieslawsoltes/AvaloniaBehaviors/blob/master/samples/BehaviorsTestApplication/Views/Pages/EventTriggerBehaviorView.axaml).
